container/go/cmd/puller/puller.go

```diff
@@ -30,11 +32,13 @@ import (
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
 )
 
 var (
 	imgName         = flag.String("name", "", "The name location including repo and digest/tag of the docker image to pull and save. Supports fully-qualified tag or digest references.")
-	directory       = flag.String("directory", "", "Where to save the images files.")
+	directory       = flag.String("directory", "", "Where to save the images files. If pulling as Docker tarball, please specify the directory to save the tarball. The tarball is named as image.tar.")
+	format          = flag.String("format", "", "The format, options being 'docker' or 'both', in which to pull the docker image. Default OCI layout format.")
```

Comment:
We should probably add a check in the main function to check the values is from {oci, docker, both} if specified. Otherwise, I can say -format=blabla, and it will select oci format. This is fine. But say I input -format=tarball, it will also download as oci.